aikiblog/forms.py

```diff
@@ -1,91 +1,130 @@
-                user.avatar = SimpleUploadedFile(avatar.name, avatar.read(), content_type='image/jpeg')
-                user.save()
-                im = Image.open(str(user.avatar))
-                a = im.size[0]
-                b = im.size[1]
-                size = (300, 300)
-                if a > b:
-                    c = (a - b)/2
-                    d = a - c
-                    box = (c, 0, d, b)
-                elif b > a:
-                    c = (b - a)/2
-                    d = b - c
-                    box = (0, c, a, d)
-                else:
-                    box = (0, 0, a, a)
-
-            region = im.crop(box)
-            region.thumbnail(size)
-            region.save(str(user.avatar), "JPEG")
-
-        if about_me:
-            user.about_text = about_me
-
-        user.save()
-
+# -*- coding: utf-8 -*-
+
+from django import forms
+from django.shortcuts import render, redirect
+from aikiblog.models import Training, User, TempAvatar
+from PIL import Image
+
+import datetime
+from annoying.functions import get_object_or_None
+from django.contrib.auth import get_user_model
+from django.core.files.uploadedfile import SimpleUploadedFile
+from os.path import join, dirname, realpath
+
+from django.http.response import HttpResponseRedirect
+
+from datetime import date
+from django.forms import widgets
+from aikiblog.models import Sensei, Dojo, TechTren
+
+CUR_DAY = datetime.datetime.now().day
+CUR_MONTH = datetime.datetime.now().month
+CUR_YEAR = datetime.datetime.now().year
+DOJOS = Dojo.objects.all
+TECHNIQUES = TechTren.objects.all
+SENSEIES = Sensei.objects.all
+
+HOUR_CHOICES = tuple(
+    [(year, year) for year in range(0, 24, +1)])
+MINUTES_CHOICES = (
+    (0, '00'),
+    (15, 15),
+    (30, 30),
+    (45, 45),
+)
+DAY_CHOICES = tuple(
+    [(year, year) for year in range(1, 32, +1)])
+MONTH_CHOICES = tuple(
+    [(year, year) for year in range(1, 13, +1)])
+YEAR_CHOICES = tuple(
+    [(year, year) for year in range(CUR_YEAR, CUR_YEAR - 30, -1)])
+dojos = Dojo.objects.all
+for dojo in dojos:
+        tup = tup+(dojo.name, dojo.name)
+
+PLACE_CHOICES = tup
+
+
+class AddTrainingForm(forms.Form):
+
+
+
+
+
+    day = forms.ChoiceField(choices=DAY_CHOICES, initial=CUR_DAY)
+    month = forms.ChoiceField(choices=MONTH_CHOICES, initial=CUR_MONTH)
+    year = forms.ChoiceField(choices=YEAR_CHOICES, initial=CUR_YEAR)
+    hour = forms.ChoiceField(choices=HOUR_CHOICES, initial=18)
+    minutes = forms.ChoiceField(choices=MINUTES_CHOICES, initial=0)
+    place = forms.ChoiceField(choices=PLACE_CHOICES)
+    #sensei = forms.ChoiceField(choices=SENSEI_CHOICES)
+    techniques = forms.ChoiceField(widget=forms.CheckboxSelectMultiple)
+    notes = forms.CharField(widget=forms.Textarea())
+
+
+
+
+
+
+
+START_YEAR_CHOICES = tuple(
+    [(year, year) for year in range(CUR_YEAR, CUR_YEAR - 30, -1)])
+K = 'K'
+M = 'M'
+SEX_CHOICES = (
+    (K, u'kobieta'),
+    (M, u'mężczyzna')
+)
+
+
+class SaveUserDataForm(forms.Form):

```

Comment:
No właśnie nie bardzo, bo nie chcę wszystkich rzeczy wyświetlać i teraz kombinuję jak to zrobić. Zrobiłem pierwszą wersję przez ModelForm, ale wtedy wyświetla mi wszystkie techniki jakie są z modelu TechTren, a trzeba je przefiltrować tak, żeby wyświetlało techniki, które dodał dany user. Musi też pobrać Usera z requesta i to są moje dwa największe pytania na dzisiaj;)